kore/src/Kore/Step/Axiom/Evaluate.hs

```diff
@@ -95,23 +117,31 @@ evaluateAxioms
         keepResultUnchanged = id
         introduceDefinedness = flip Pattern.andCondition
         markRemainderEvaluated = fmap TermLike.mkEvaluated
+        rules = Result.appliedRule <$> Result.results result
 
     simplifiedResults <-
-        OrPattern.simplifyPredicatesWithSmt
-           predicate (Step.gatherResults result)
+        traverse (OrPattern.simplifyPredicatesWithSmt predicate)
+            ((fmap Result.result . Result.results) result)
 
     simplifiedRemainders <-
```

Comment:
I know you didn't change this part of the code, but the same comment applies as just above: 
```suggestion
    simplifiedRemainders <- Lens.traverseOf (field @"remainders") (OrPattern.simplifyPredicatesWithSmt predicate) simplifiedResults
```